task_generator/task_generator/manager/entity_manager/__init__.py

```diff
@@ -0,0 +1,133 @@
+        Loads given obstacles into the simulator. 
+        If the object has an interaction radius of > 0, 
+        then load it as an interactive obstacle instead of static
+        """
+        raise NotImplementedError()
+
+    def spawn_dynamic_obstacles(self, obstacles: Collection[DynamicObstacle]):
+        """
+        Loads given obstacles into the simulator.
+        Currently by loading a existing sdf file, 
+        then reaplacing the static values by dynamic ones 
+        """
+        raise NotImplementedError()
+
+    def spawn_walls(self, walls: WorldWalls, heightmap: WorldMap):
+        """
+        Adds walls to the simulator.
+        """
+        raise NotImplementedError()
+
+    def unuse_obstacles(self):
+        """
+        Prepares obstacles for reuse or removal.
+        """
+        raise NotImplementedError()
+
+    def remove_obstacles(self, purge: ObstacleLayer = ObstacleLayer.UNUSED):
+        """
+        Removes obstacles from simulator.
+        @purge: if False, only remove unused obstacles
+        """
+        raise NotImplementedError()
+
+    def spawn_robot(self, robot: Robot):
+        """
+        Spawns a robot.
+        @robot: Robot description.
+        """
+        raise NotImplementedError()
+
+    def move_robot(self, name: str, position: PositionOrientation):
+        """
+        Moves a robot.
+        @name: Robot name
+        @position: Target position
+        """
+        raise NotImplementedError()
+
+
+EntityManagerRegistry = Registry[Constants.EntityManager, EntityManager]()
+
+
+@EntityManagerRegistry.register(Constants.EntityManager.DUMMY)
+def dummy():
+
+    class DummyEntityManager(EntityManager):
+        def __init__(self, namespace: Namespace, simulator: BaseSimulator):
+            super().__init__(namespace, simulator)
+            self.__logger = self._node.get_logger().get_child('dummy_EM')
+
+        def spawn_obstacles(self, obstacles: Collection[Obstacle]):
+            self.__logger.debug(f'spawning {len(obstacles)} static obstacles')
+
+        def spawn_dynamic_obstacles(self, obstacles: Collection[DynamicObstacle]):
+            self.__logger.debug(f'spawning {len(obstacles)} dynamic obstacles')
+
+        def spawn_walls(self, walls: WorldWalls, heightmap: WorldMap):
+            self.__logger.debug(f'spawning {len(walls)} walls')
+
+        def unuse_obstacles(self):
+            self.__logger.debug(f'unusing obstacles')
+
+        def remove_obstacles(self, purge: ObstacleLayer = ObstacleLayer.UNUSED):
+            self.__logger.debug(f'removing obstacles (level {purge})')
+
+        def spawn_robot(self, robot: Robot):
+            self.__logger.debug(f'spawning robot {Robot.name}')
+
+        def move_robot(self, name: str, position: PositionOrientation):
+            self.__logger.debug(
+                f'moving robot {Robot.name} to {repr(position)}')
+
+    return DummyEntityManager
+
+# @EntityManagerRegistry.register(Constants.EntityManager.HUNAVSIM)
```

Comment:
@AhmedMartban Is this compatible with your draft? Ideally, you'd only need to change your EntityManager import (remove the last .entity_manager).

Reply:
Yes works fine now thank you!